nomenclature/testing.py

```diff
@@ -55,6 +56,8 @@ def assert_valid_structure(
         - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
 
     """
+    if dimensions == []:
```

Comment:
Together with my comment above I would go with the following:
```suggestion
    if dimensions is None:
```